test/functional/plugin/lsp_spec.lua

```diff
@@ -304,6 +309,7 @@ describe('LSP', function()
       }
     end)
     it('workspace/configuration returns NIL per section if client was started without config.settings', function()
+      clear()
       fake_lsp_server_setup('workspace/configuration no settings')
```

Comment:
Just curious, why did you directly use the fake_lsp_server_setup instead of the test_rpc_server (which handles stop) here?

Reply:
The main intention was to reduce the boilerplate a bit and keep the test focused on the given scenario.

But the changes here in the PR so far don't work, so maybe using `test_rpc_server` is the way to go. At least one commit confirms that removing the test case completely fixes the linux asan tests.

I'll give it another go a bit later

Reply:
Your call! I personally leans towards just using the test RPC class for consistency

Reply:
I agree with @mjlbach

Reply:
I had another look at this, but using `test_rpc_server` would require extending the tests in `fake-lsp-server` as well.
To me it seems a bit overkill to emulate the servers flow when all I want is to test the `workspace/configuration` callback in isolation.

I'd rather keep it as it is and work towards simplifying some of the others as well where the full blown `test_rpc_server` is not required. Or make `test_rpc_server` easier to use.

Reply:
ok thanks for the explanation. Going to merge to fix ci.